### ISSUE TITLE
Package mg5amcnlo dependencies

### ISSUE THREAD

**Issue by matthewfeickert** [hep-packaging-coordination/packaging-hep-simulation-stack#3]
As noted in https://github.com/mg5amcnlo/mg5amcnlo/issues/156 mg5amcnlo vendors lots of software without including their licenses and also has multiple external dependencies. 

All of these would need to be packaged on conda-forge to realistically package `mg5amcnlo`.

* conda-forge: https://github.com/conda-forge/mg5amcnlo-feedstock
* [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/mg5amcnlo.svg)](https://anaconda.org/conda-forge/mg5amcnlo)

## Vendored code

* [x] CutTools
   - website: https://www.ugr.es/~pittau/CutTools/
   - conda-forge: https://github.com/conda-forge/cuttools-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/cuttools-static.svg)](https://anaconda.org/conda-forge/cuttools-static)
   - Requires:
      - [x] QCDLoop (Fortran version)
      - [x] avh (through OneLOop)
      - [x] mpfun90
      - Currently has no license (emailed Roberto to ask him to add a license on 2025-02-03)
* [ ] DiscreteSampler
   - source: https://github.com/mg5amcnlo/mg5amcnlo/blob/55a291dc8def762938a964cc1343c6d76af5b843/vendor/DiscreteSampler/ (I don't think it exists anywhere else) 
* [x] IREGI
   - source: https://github.com/mg5amcnlo/mg5amcnlo/tree/55a291dc8def762938a964cc1343c6d76af5b843/vendor/IREGI/src (I don't think it exists anywhere else)
   - conda-forge: https://github.com/conda-forge/iregi-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/iregi-static.svg)](https://anaconda.org/conda-forge/iregi-static)
* [ ] StdHEP
   - source: https://github.com/mg5amcnlo/mg5amcnlo/tree/55a291dc8def762938a964cc1343c6d76af5b843/vendor/StdHEP (I don't think it exists anywhere else)
* [ ] SudGen
   - source: https://github.com/mg5amcnlo/mg5amcnlo/tree/55a291dc8def762938a964cc1343c6d76af5b843/vendor/SudGen (I don't think it exists anywhere else)
* [ ] eepdfgrid
   - source: https://github.com/mg5amcnlo/mg5amcnlo/tree/55a291dc8def762938a964cc1343c6d76af5b843/vendor/eepdfgrid (I don't think it exists anywhere else)
* [x] ply (though mg5amcnlo modifies it for unknown reasons: https://github.com/mg5amcnlo/mg5amcnlo/commits/55a291dc8def762938a964cc1343c6d76af5b843/vendor/ply)
   - website: https://www.dabeaz.com/ply/index.html
   - GitHub: https://github.com/dabeaz/ply
   - conda-forge: https://github.com/conda-forge/ply-feedstock/
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/ply.svg)](https://anaconda.org/conda-forge/ply)
* [x] OneLOop
   - website: https://helac-phegas.web.cern.ch/OneLOop.html
   - Bitbucket: https://bitbucket.org/hameren/oneloop/
   - conda-forge: https://github.com/conda-forge/oneloop-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/oneloop.svg)](https://anaconda.org/conda-forge/oneloop)

## External dependencies

From https://github.com/mg5amcnlo/HEPToolsInstallers:

* [x] COLLIER
   - website: https://collier.hepforge.org/
   - code releases: https://collier.hepforge.org/downloads/
   - conda-forge: https://github.com/conda-forge/collier-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/collier.svg)](https://anaconda.org/conda-forge/collier)
* [x] EMELA
   - GitHub: https://github.com/gstagnit/eMELA/
   - conda-forge: https://github.com/conda-forge/emela-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/emela.svg)](https://anaconda.org/conda-forge/emela)
* [ ] ginac_mg5_interface
* [ ] MadAnalysis5
   - GitHub: https://github.com/MadAnalysis/madanalysis5
* [ ] NKLO_TOOLS
* [x] Ninja (not the build tool)
   - website: https://ninja.hepforge.org/
   - GitHub: https://github.com/peraro/ninja
   - conda-forge: https://github.com/conda-forge/ninja-hep-ph-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/ninja-hep-ph.svg)](https://anaconda.org/conda-forge/ninja-hep-ph) (use of `libquadmath` disallows macOS)
   - Requies:
      - [x] OneLOop
      - [x] LoopTools
* [x] RIVET
   - website: https://rivet.hepforge.org/
   - GitLab: https://gitlab.com/hepcedar/rivet
   - conda-forge: https://github.com/conda-forge/rivet-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/rivet.svg)](https://anaconda.org/conda-forge/rivet)
   - [Requires](https://gitlab.com/hepcedar/rivetbootstrap/-/blob/main/rivet-bootstrap?ref_type=heads):
      - [x] HepMC3
      - [x] FastJet
      - [x] `fjcontrib`
      - [x] [HDF5](https://github.com/conda-forge/hdf5-feedstock/)
      - [x] [HighFive](https://github.com/conda-forge/highfive-feedstock/)
      - [x] YODA
      - [x] [Cython](https://github.com/conda-forge/cython-feedstock/)
      - [x] [`pyyaml`](https://github.com/conda-forge/pyyaml-feedstock/)
      - [x] [`requests`](https://github.com/conda-forge/requests-feedstock/)
      - [x] [`matplotlib`](https://github.com/conda-forge/matplotlib-feedstock/)
      - [x] [`tqdm`](https://github.com/conda-forge/tqdm-feedstock/)
* [ ] contur
   - GitLab: https://gitlab.com/hepcedar/contur
* [ ] dragon
   - GitHub: https://github.com/cosmicrays/DRAGON
* [ ] dragon data
   - source: http://madgraph.phys.ucl.ac.be/Downloads/maddm/dragon_input_from_galprop.zip
* [x] fitsio
   - source: https://heasarc.gsfc.nasa.gov/FTP/software/fitsio/c/
   - GitHub: https://github.com/esheldon/fitsio
   - conda-forge: https://github.com/conda-forge/fitsio-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/fitsio.svg)](https://anaconda.org/conda-forge/fitsio)
* [x] gsl
   - website: https://www.gnu.org/software/gsl/
   - conda-forge: https://github.com/conda-forge/gsl-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/gsl.svg)](https://anaconda.org/conda-forge/gsl)

## Dependencies of dependencies

* [x] FF
   - website: https://www.nikhef.nl/~form/maindir/others/ff/ff.html
   - GitHub: https://github.com/hep-packaging-coordination/ff
   - https://github.com/hep-packaging-coordination/packaging-hep-simulation-stack/issues/4
   - Blocked by:
      - [x] https://github.com/hep-packaging-coordination/ff/issues/4
   - conda-forge: https://github.com/conda-forge/ff-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/ff-static.svg)](https://anaconda.org/conda-forge/ff-static)
* [x] LoopTools
   - website: https://feynarts.de/looptools/
   - Requires: FF
   - conda-forge: https://github.com/conda-forge/looptools-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/looptools-static.svg)](https://anaconda.org/conda-forge/looptools-static)
* [x] YODA
   - conda-forge: https://github.com/conda-forge/yoda-feedstock/
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/yoda.svg)](https://anaconda.org/conda-forge/yoda)
* [x] QCDLoop (Fortran version named `ql`)
   - website: https://qcdloop.fnal.gov/
   - conda-forge: https://github.com/conda-forge/qcdloop-fortran-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/qcdloop-fortran-static.svg)](https://anaconda.org/conda-forge/qcdloop-fortran-static) 
   - Requires:
      * FF
      * Currently has no license (on 2025-02-03 got permission to license as MIT)
* [x] QCDLoop (C++ version)
   - website: https://qcdloop.web.cern.ch/
   - GitHub: https://github.com/scarrazza/qcdloop
   - conda-forge: https://github.com/conda-forge/qcdloop-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/qcdloop.svg)](https://anaconda.org/conda-forge/qcdloop) (use of `libquadmath` disallows macOS)
* [x] MPFUN90
   - website: https://www.davidhbailey.com/dhbsoftware/ (doesn't have a MPFUN90 distribution on it)
   - GitHub: https://github.com/APN-Pucky/mpfun90
   - conda-forge: https://github.com/conda-forge/mpfun90-feedstock
   - [![Conda Platforms](https://img.shields.io/conda/pn/conda-forge/mpfun90.svg)](https://anaconda.org/conda-forge/mpfun90)

**Comment by APN-Pucky** [comment]
FYI, gentoo science has cuttools split of for openloops, mcfm, ... https://github.com/gentoo/sci/blob/master/sci-physics/cuttools/cuttools-1.9.3-r1.ebuild
It is a bit of a mess to get unbundle included libraries. Another similar hep packaging project to this is https://github.com/andriish/HEPrpms if you need a reference for some package.

**Comment by APN-Pucky** [comment]
I am curious how you want to get madgraph to run the first initial required run? Is it something that conda supports?

> MadGraph needs to generate `Template/LO/Source/make_opts` which is done
    automatically at startup.  This needs to be done during setup (or with
    root access)
    
 from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=madgraph and gentoo does something similar https://github.com/gentoo/sci/blob/20a07b35603efba2ab1ac0310e4604506f51ed91/sci-physics/madgraph5/madgraph5-3.5.3.ebuild#L76 or is this no longer needed (there was some effort to make madgraph more packaging friendly, but i don't know of the outcome)

**Comment by matthewfeickert** [comment]
> I am curious how you want to get madgraph to run the first initial required run? Is it something that conda supports?
> 
> > MadGraph needs to generate `Template/LO/Source/make_opts` which is done
> > automatically at startup.  This needs to be done during setup (or with
> > root access)
> 
> from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=madgraph

@APN-Pucky When I containerized it a few years ago (https://github.com/scailfin/MadGraph5_aMC-NLO) I [just did](https://github.com/scailfin/MadGraph5_aMC-NLO/blob/ba3efc64530e192765abd07cd4988f22b075b496/docker/centos/Dockerfile#L243)

```
echo "exit" | mg5_aMC
```

and while I haven't thought about this much I would assume that in the `build` stage for conda-forge I would just make sure that the required directory structure existed under `$PREFIX` such that `$PREFIX/MG5_aMC/bin/mg5_aMC` existed (and would be symlinked to `$PREFIX/bin/mg5_aMC`) and then just do

```
echo "exit" | $PREFIX/bin/mg5_aMC
```

so that the `$PREFIX/MG5_aMC/Template/` directory structure existed. `conda-build` knows how to go through, discover, and correct all the paths of everything under `$PREFIX` so you don't have to manually do that to make sure that the paths in `Template/LO/Source/make_opts` are pointing to paths on the build machine that don't exist.
